### PR TITLE
Code cleanup.

### PR DESCRIPTION
**Pull request by Ascor8522**
I know this commit might look intimidating, but really, it's not.

Since we are using Node.js 16 and above, we might as well use modern features:
- turned the main file into a module, so we can use top level await
- we don't need to use an IIFE to await async code anymore
- turned variables into proper functions
- removed un-necessary await everywhere, since everything is now one long and clean promise chain
- removed the un-necessary try-catch statements, since errors in promises can be caught once at the end of the chain
- added some checks after fetches for extra safety

Also:
- Added documentation to functions and some parameter / return types with JSDoc.
- Created a config object at the top of the file. An .env file could have been used too.
- Made code more easily understandable, split it into smaller function with more descriptive names.
- GitPod file no longer needed, just use npx command to run a cli script from installed packages.
- Added Node.js >=16.0.0 as a requirement to the npm package.
- Removed un-needed packages (fs-extra) since we were only using 1 function that can be re-written easily.
- Added tsconfig file so VSCode can automatically check the file with the builtin TypeScript compiler.

### CODE REVIEW

**Comment by Hans5958** [comment]
I don't think that this is scary or whatnot. I can see that you are someone who is bored and, with the knowledge of TypeScript or some cutting-edge technologies, scrolled through the list of the org's repository and try to improve one of it. Likewise, I applaud your undertaking.

Let me tell you that I know enough about TypeScript, and much of JavaScript, that I needed to make things works. I just make things that even though it is not effective, it is enough to make things work. New technologies are interesting, even simple things like `const` and `let`, but I didn't try to succumb myself with it and just try to make real things. 

This means that I could just use TypeScript since the start, but I decided not to because a simple Node.js is enough to make it work. And also I need to do many searches to understand what is going on. All of it ends with "oh, that's cool."

There is one criticism, though. It is the fact you "split [the] into smaller function with more descriptive names." Modularization is good, don't get me wrong, but there must be a limit that too much modularization is just bad and unnecessary, in respect of the complexity of a project. It's hard to explain, so I think [this video](https://www.youtube.com/watch?v=QM1iUe6IofM) would explain it better.

This PR also makes me remember a simple thing, to stop leaving things out of the blue and start commenting! I'm not just talking about assigning types with JSDoc, but with simple comments like `I did this because it is better and nothing is going to stop me` or `This does this and please don't touch it due to the flexibility of the dataset`.

Some other comments:
- I would prefer functions positioned work from top to bottom, and then ended with the execution on the bottom.
- I would prefer that we don't have such thing as zero commits, It's fine that this field is missing completely (this is how I coded on the contributors page, because zero means that it is shown, but 0), but I can see why you prefer consistency. If you really, really needed it, just do `undefined` or `null`.
- I prefer doing things with `await` than the `.then()` chain. I think it is as weird as a `if` chain, but I think this is just an opinion of mine.

That being said, I will try to come back to this in the few days, because I still need to focus on the website. Once again, I applaud your hard work on this repository, even though it is still unused until that PR got merged on the website.

**Review by Hans5958 — APPROVED** [review]
I don't think that this is scary or whatnot. I can see that you are someone who is bored and, with the knowledge of TypeScript or some cutting-edge technologies, scrolled through the list of the org's repository and try to improve one of it. Likewise, I applaud your undertaking.

Let me tell you that I know enough about TypeScript, and much of JavaScript, that I needed to make things works. I just make things that even though it is not effective, it is enough to make things work. New technologies are interesting, even simple things like `const` and `let`, but I didn't try to succumb myself with it and just try to make real things. 

This means that I could just use TypeScript since the start, but I decided not to because a simple Node.js is enough to make it work. And also I need to do many searches to understand what is going on. All of it ends with "oh, that's cool."

There is one criticism, though. It is the fact you "split [the] into smaller function with more descriptive names." Modularization is good, don't get me wrong, but there must be a limit that too much modularization is just bad and unnecessary, in respect of the complexity of a project. It's hard to explain, so I think [this video](https://www.youtube.com/watch?v=QM1iUe6IofM) would explain it better.

This PR also makes me remember a simple thing, to stop leaving things out of the blue and start commenting! I'm not just talking about assigning types with JSDoc, but with simple comments like `I did this because it is better and nothing is going to stop me` or `This does this and please don't touch it due to the flexibility of the dataset`.

Some other comments:
- I would prefer functions positioned work from top to bottom, and then ended with the execution on the bottom.
- I would prefer that we don't have such thing as zero commits, It's fine that this field is missing completely (this is how I coded on the contributors page, because zero means that it is shown, but 0), but I can see why you prefer consistency. If you really, really needed it, just do `undefined` or `null`.
- I prefer doing things with `await` than the `.then()` chain. I think it is as weird as a `if` chain, but I think this is just an opinion of mine.

That being said, I will try to come back to this in the few days, because I still need to focus on the website. Once again, I applaud your hard work on this repository, even though it is still unused until that PR got merged on the website.

File: package.json
Comment (RedGuy12) [inline package.json:11]: the `type: module` makes the `.mjs` extension unnecessary

Comment (RedGuy12) [reply package.json:11]: But `"type": "module"` makes it a module, and so does `type="module"` on `<script>` tags in the browser

File: src/types.mjs
Comment (RedGuy12) [inline src/types.mjs:3]: You can use .d.ts files even when using JSDoc...

File: package.json
Comment (Hans5958) [reply package.json:11]: > Except you need to use modules (thus with the .mjs extension) in the browser (this is a browser requirement) to use some features like top level await, which is quite convenient and that I chose to use.

Fact is, this script is ran on Node.js, so `.mjs` is unnecessary.

**Comment by Hans5958** [comment]
By the way, I see that you reacted to my message, but you haven't addressed my concerns. I may need to do something on it, either closing it and save it for tommorow, or edit it but I have limited time, or you can do it.

**Review by Hans5958 — CHANGES_REQUESTED** [review]
By the way, I see that you reacted to my message, but you haven't addressed my concerns. I may need to do something on it, either closing it and save it for tommorow, or edit it but I have limited time, or you can do it.

File: package.json
Comment (Ascor8522) [reply package.json:11]: > > Except you need to use modules (thus with the .mjs extension) in the browser (this is a browser requirement) to use some features like top level await, which is quite convenient and that I chose to use.
> 
> Fact is, this script runs on Node.js only, so `.mjs` is unnecessary.

Well, I don't know what I had i mind when writing my previous answer but yeah, this code is meant to run on node.

Anyway, to use top level await in node (14+), you still need to use ES modules. This can be achieved, as mentionned, in multiple ways:
* either by using the `.mjd` extension (to tell the module files appart from the regular JS),
* or by setting the `type` property to `module` in the `package.json`.

I chose to do both (which, by the way, doesn't hurt), to make it clear that we are actually dealing with modules. This way, you don't have to look in the `package.json` file for a detail you usually wouldn't pay attention to and that you probably would have missed. The `.mjs` extension makes it more noticeable. Also, since node and other services check the `package.json` file for several reasons, we might as well make it clear there too that we are working with ES modules.

File: src/types.mjs
Comment (Ascor8522) [reply src/types.mjs:3]: Sure, but note that, this too, is only considered as a step towards transitionning to TypeScript.
It is even mentionned on the homepage of the TS website: from JS, to JS + JSDoc, to JS + typings (`.d.ts`), to finally TS.

And since we are only working with JS yet, I preferred to keep it coherent with the codebase (in JS). That makes it easier for JS-only devs to work with, without scaring them with TypeScript or typings files at all.

**Comment by Ascor8522** [comment]
> By the way, I see that you reacted to my message, but you haven't addressed my concerns. I may need to do something on it, either closing it and save it for tommorow, or edit it but I have limited time, or you can do it.

It will probably sound quite obnoxious, so in adavance, I'm sorry, but I don't really get most of the reasons why wou'd want to have the code changed, other than the 0 commit stuff. But anyway, you "own" the code and do the merges...

* That function order is quite common for JavaScript. The entry point comes first, since, well, it's the entry point (the start is meant fot the start, and the end for the end). The other functions are just written in the order they are called. Declaring the utility functions first wouldn't really be helpfull, since you'd still need to look at the entry point to see in what order they are called to fully understand the code. Maybe you're just used to another way of doing things, which, isn't bad or anything, but just different. I know this is the case in Python for example, and I saw you do some Python as well... But I'm just guessing at this point.
* About the zero commits, I do understand that it might seem odd. But what was the original behavior? It's pretty hard to guess from the actual code what it does and there is no explanation or comments. Do we just hide those users at all or do we set their number of commits to null (or even better, make that propery optional?) and it's won't show the exact number in the rendered document?
* The `await` vs `.then()` chain just sounds like a syntax preference to me, and doesn't add much. I usually prefer to work with  `.then()` so that I don't have to create temporary (understand "avoidable" and "noisy") variables in the scope, just for those to only be used on the next line and nowhere else. Passing the result of a function as the parameter of the next one is, I have to admit, quite convenient, and eliminates that need. Is it worth rewriting the whole function just for that sole reason? I don't think so, but again, you have the final word.
* Finally, when it comes to using code in a non-standard (proprietery) environement, I usually avoid adding (commiting) those additional files, since they bring no value for the other users. (See I didn't add my `.vscode` folder, nor my Intellij `.idea` folder, nor my Eclipse, no my NetBeans, nor any other editor's config file I use.) Even more when this can be replaced with a simple command that would work for all users and that can also be put (and thus "documented") in the `package.json` file. Come one, that command only takes half a second to be typed in the terminal, and sometimes some editors will create shortcuts or buttons to run those commands more easily. Also, just out of curiosity, why do you use GitPod, especially since VSCode is now available in all major web browsers, and can be opened in any GitHub repo directly by pressing the "." (dot) key on your keyboard.

Hope this wan't too much reading for just a couple of personal opions and a question ^^'

I'll change the 0 commits thingy as soon as I get your response.

Thanks again for taking the time to review this, really appreciate 👍

File: src/types.mjs
Comment (RedGuy12) [reply src/types.mjs:3]: I get that, although I feel like longer types in JSDoc just get messy

File: package.json
Comment (CST1229) [reply package.json:11]: > either by using the `.mjd` extension (to tell the module files appart from the regular JS),

Isn't it `.mjs`?

**Comment by Hans5958** [comment]
Sorry for the wait. Exams are closer, and I'm quite busy these days.

I understand that most of this is just style preferences, but here's my thought of it:

- To be fair, I'm the one who prefers definitions on top, and execution on the bottom. Doing it on the top can confuse people when scrolling, whilst when I put it on the end, I know the end of the file is the end of the execution. With this, I also prefer that execution goes from top to bottom, and so does what the code should be. I'm quite tired of having hoisting on var and function in the past, so I think it is a step forward.

- About `await`, it is just my preference with the same basis as why you should avoid if-else chain. If I can remove an indent, why the heck not? Noise shouldn't matter, especially in JS where we have automatic garbage collectors, if that's a thing in JS.

- Believe it or not, I also never push configurations on repositories. Yes, gitpod.yml is an exception, because what I was doing on Gitpod is to add contributors with the tool, and that requires Node.js. Obviously, last time, we don't have `github.dev` and Codespaces. In fact, you should compare it to Codespaces in this case. I would like to add a `devcontainer.json`, but I decided not today. Last time my laptop is a potato, but this time it is a relative beast. But, hey, I hope some people can find it useful, even if I don't need it anymore, somehow.

About the commit, the current behavior is to not show the commit count if it is undefined/null. Yes, it may be impossible for it to be 0, but I just want to do it truthly, if you know what I meant. Yes, this is not mentioned on the files, hence I told you about my minimal comments on this repo.

I guess that's it for now. LMK if you need me.

**Pull request closed without merging**
